spec/v0.4.1-draft/spec.md

```diff
@@ -578,67 +577,48 @@ In the remainder of this section, the current list of relation extension types (
 
-A set has the following properties in its `data` object:
+- Name: `@ocif/rel/group`
+- URI: `https://spec.canvasprotocol.org/v0.4.1/core/group-rel.json`
 
-| Property  | JSON Type | OCIF Type   | Required     | Contents                  |
-| --------- | --------- | ----------- | ------------ | ------------------------- |
-| `members` | `array`   | [ID](#id)[] | **required** | IDs of members of the set |
+A group relation is a relation which groups nodes together.
+Groups are known as "Groups" in most canvas apps,
+"Groups" in Godot, and "Tags" in Unity.
 
-- **members**: A list of IDs of nodes or relations that are part of the set.
-  Resources cannot be part of a set.
+A group has the following properties in its `data` object:
 
-**Example:** A set relation with three members:
+| Property        | JSON Type | OCIF Type   | Required     | Contents                    |
+| --------------- | --------- | ----------- | ------------ | --------------------------- |
+| `members`       | `array`   | [ID](#id)[] | **required** | IDs of members of the group |
+| `cascadeDelete` | `boolean` | `boolean`   | **optional** | `true` or `false`           |
 
-```json
-{
-  "type": "@ocif/set",
-  "members": ["n1", "n2", "n3"]
-}
-```
+- **members**: A list of IDs of nodes or other groups that are part of the group.
+  Resources cannot be part of a group.
+- **cascadeDelete**: A boolean flag indicating if deleting the group should also delete all members of the group.
+  If `true`, deleting the group will also delete all members of the group.
+  If `false`, deleting the group will not delete its members.
 
-**Example:** A node using the set relation would look like this:
+**Example:** A group of 3 nodes with letters for names:
 
 ```json
 {
-  "id": "nodeA",
+  "id": "letter_named_nodes",
   "data": [
     {
-      "type": "@ocif/set",
-      "members": ["n1", "n2", "n3"]
+      "type": "@ocif/rel/group",
+      "members": [
+        "A",
+        "B",
+        "C"
+      ]
     }
   ]
 }
 ```
 
-JSON schema: [set-rel.json](core/set-rel.json)
-
-## Group Relation
-
-- Name: `@ocif/rel/group`
-- URI: `https://spec.canvasprotocol.org/v0.4.1/core/group-rel.json`
-
-A group relation is a relation, which groups nodes together.
-It implies stronger semantics than a [set relation](#set-relation).
-
-A group is modeled as a relation with a list of its members.
-
-| Property  | JSON Type | OCIF Type   | Required     | Contents                  |
-| --------- | --------- | ----------- | ------------ | ------------------------- |
-| `members` | `array`   | [ID](#id)[] | **required** | IDs of members of the set |
-
-- **members**: A list of IDs of nodes (or relations, such as other groups) that are part of the group.
-
-**Semantics**
-
 - Groups can contain groups as members. Thus, all semantics apply recursively.
-- When a group is deleted, all members are deleted as well.
+- When a group is deleted, if `"cascadeDelete"` is `true`, all members are deleted as well.
```

Comment:
If cascadeDelete is true on group A which contains a group B, and group B has cascadeDelete=false, are members of B deleted, when A is deleted? I would say, no.

Reply:
I think that is implied from the existing text, because the operation on A is not "recursiveDelete", and the operation of deleting B doesn't cascade by definition. Group A doesn't know about the contents of group B. But maybe we should clarify this if the text is unclear.